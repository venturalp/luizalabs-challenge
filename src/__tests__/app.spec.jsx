import React from 'react'
import App from '../app'
import { renderWithRouter, ThemeWrapper } from 'Commons/tests/Tests.Helpers'
import { renderHook, act } from '@testing-library/react-hooks'
import { useApplicationStore } from 'Modules/application/Application.Store'
import { screen, waitFor } from '@testing-library/dom'

describe('test app', () => {
  it('render test app', () => {
    const { asFragment } = renderWithRouter(
      <ThemeWrapper>
        {' '}
        <App />
      </ThemeWrapper>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('render test app with loading', async () => {
    const { result } = renderHook(() => useApplicationStore())
    act(() => {
      result.current.setIsLoading(true)
    })
    renderWithRouter(
      <ThemeWrapper>
        {' '}
        <App />
      </ThemeWrapper>,
    )
    await waitFor(async () => {
      expect(await screen.getByTestId('loadingComponent')).toBeInTheDocument()
    })
  })
})
