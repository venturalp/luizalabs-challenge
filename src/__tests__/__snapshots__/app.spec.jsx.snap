// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app render test app 1`] = `
<DocumentFragment>
   
  <div
    class="sc-gsTCUz bGKjuf"
  >
    Home page
  </div>
</DocumentFragment>
`;
